pandas/core/frame.py

```diff
@@ -2954,7 +2954,7 @@ def __getitem__(self, key):
             return self.where(key)
 
         # Do we have a (boolean) 1d indexer?
-        if com.is_bool_indexer(key):
+        if com.is_bool_indexer(key) or isinstance(key, Series):
```

Comment:
should _any_ Series go through here?   e.g. what if it is Series[dt64] instead of Series[bool]?  should it only be empty Series?

Reply:
Only empty